### ISSUE TITLE
buffer_search::Dismiss not working in Editor context

### ISSUE THREAD

**Issue by networkhermit** [zed-industries/zed#4564]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I tried to use `ctrl-l` to dismiss the buffer_search, but it seems not working in the `Editor` context. When I replace the keybinding to other command, it works as expected.

```json
  {
    "bindings": {
      "ctrl-l": "buffer_search::Dismiss"
    },
    "context": "Editor"
  }
```

### Environment

Zed: v0.96.4 (stable)
OS: macOS 13.4.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by JosephTLyons** [comment]
Hey @networkhermit, not sure if you are still using Zed or not, but if so, would you be able to check if this is still an issue for you in current Zed?

**Comment by networkhermit** [comment]
Yes. This issue still persists in the latest stable release Zed 0.145.1.

> When the focus is in editor buffer, buffer_search::Dismiss doesn't work. This can be reproduced if you use * to search word under cursor in vim mode, or make sure your cursor is in text buffer (not the search box) if you don't use vim mode.

```json
{
  "vim_mode": true
}
```

Neither of the following keybinding works.

```json
[
  {
    "bindings": {
      "ctrl-l": "buffer_search::Dismiss"
    },
    "context": "Editor"
  }
]
```

```json
[
  {
    "bindings": {
      "ctrl-l": "buffer_search::Dismiss"
    },
    "context": "BufferSearchBar"
  }
]
```

**Comment by notpeter** [comment]
I am able to reproduce this without custom keybinds:

1. Open a buffer (without buffer search deployed)
2. Deploy the buffer search (e.g. `cmd-f` or `buffer search: deploy from palette) so it is in focus.
3. Open the command palette, search for `buffer` and note `buffer search: dismiss` is present.
4. Click to the editor so it's in focus.
5. Open the command palette, search for `buffer` and note `buffer search: dismiss` is absent.

The `buffer_search::Dismiss` is unavailable from the `Editor` context.
